pkg/apis/infinispan/v1/types_util.go

```diff
@@ -365,7 +370,7 @@ func (ispn *Infinispan) IsClientCertEnabled() bool {
 
 // IsGeneratedSecret verifies that the Secret should be generated by the controller
 func (ispn *Infinispan) IsGeneratedSecret() bool {
-	return ispn.IsAuthenticationEnabled() && (ispn.Spec.Security.EndpointSecretName == "" || strings.HasSuffix(ispn.Spec.Security.EndpointSecretName, consts.GeneratedSecretSuffix))
+	return strings.HasSuffix(ispn.Spec.Security.EndpointSecretName, consts.GeneratedSecretSuffix)
```

Comment:
in this case we could validate whole secret name, not just suffix to avoid confusion